core/module_specifier.rs

```diff
@@ -1,6 +1,67 @@
+use std::error::Error;
 use std::fmt;
+use std::path::Component;
+use std::path::Path;
+use std::path::PathBuf;
+use std::path::Prefix;
+use url::ParseError;
 use url::Url;
 
+/// Error indicating the reason resolving a module specifier failed.
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+pub enum ResolveError {
+  InvalidUrl(ParseError),
+  InvalidBaseUrl(ParseError),
+  UnqualifiedRelativePath,
+  UnsupportedPathPrefix,
+}
+
+impl Error for ResolveError {
+  fn source(&self) -> Option<&(dyn Error + 'static)> {
+    use ResolveError::*;
+    match self {
+      InvalidUrl(ref err) => Some(err),
+      InvalidBaseUrl(ref err) => Some(err),
+      _ => None,
+    }
+  }
+}
+
+impl fmt::Display for ResolveError {
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    use ResolveError::*;
+    match self {
+      InvalidUrl(ref err) => write!(f, "invalid URL: {}", err),
+      InvalidBaseUrl(ref err) => {
+        write!(f, "invalid base URL for relative import: {}", err)
+      }
+      UnqualifiedRelativePath => {
+        write!(f, "relative import path must start with / or ./ or ../")
+      }
+      UnsupportedPathPrefix => {
+        write!(f, "invalid path prefix, must be disk prefix (eg. C:/)")
```

Comment:
Is just `/` or `\\` also acceptable?
I think this error may be a bit confusing for non-windows users.

Reply:
Both are acceptable. Let's change to: `invalid path prefix, must be disk prefix (eg. C:)`?